Src/WitsmlExplorer.Api/Workers/AnalyzeGapWorker.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        return GetGapReportResult(job, gapReportItems, isDepthLog, logUid);
+    }
+
+    private static Func<Index, bool> ValidateCurveIndex(bool isLogIncreasing, LogCurveIndex logCurveIndex)
+    {
+        return isLogIncreasing
+            ? value => value >= logCurveIndex.MinIndex && value <= logCurveIndex.MaxIndex
+            : value => value <= logCurveIndex.MinIndex && value >= logCurveIndex.MaxIndex;
+    }
+
+    private static Index GetDepthOrDateTimeIndex(bool isDepthLog, string value, LogCurveIndex logCurveIndex)
+    {
+        return isDepthLog && logCurveIndex.MinIndex is DepthIndex depthMinIndex
+            ? new DepthIndex(double.Parse(value, CultureInfo.InvariantCulture), (depthMinIndex).Uom)
+            : new DateTimeIndex(DateTime.Parse(value, CultureInfo.InvariantCulture));
+    }
+
+    private (WorkerResult, RefreshAction) GetGapReportResult(AnalyzeGapJob job, List<AnalyzeGapReportItem> gapReportItems, bool isDepth, string logUid)
+    {
+        Logger.LogInformation("Analyzing gaps is done. {jobDescription}", job.Description());
+        job.JobInfo.Report = GetGapReport(gapReportItems, job.LogReference, isDepth);
+        WorkerResult workerResult = new(GetTargetWitsmlClientOrThrow().GetServerHostname(), true, $"Analyze gaps for log: {logUid}", jobId: job.JobInfo.Id);
+        return (workerResult, null);
+    }
+
+    private AnalyzeGapReport GetGapReport(List<AnalyzeGapReportItem> analyzeGapItems, LogObject logReference, bool isDepthLog)
+    {
+        return new AnalyzeGapReport
+        {
+            Title = $"Analyze gaps report",
+            Summary = analyzeGapItems.Count > 0
+                ? $"Found {analyzeGapItems.Count} gaps for {(isDepthLog ? "depth" : "time")} log '{logReference.Name}':"
+                : "No gaps were found for selected items.",
+            LogReference = logReference,
+            ReportItems = analyzeGapItems
+        };
+    }
+
+    /// <summary>
+    /// Get all gaps for input list of mnemonic values larger then requested gap size.
+    /// </summary>
+    /// <param name="mnemonic">Name of mnemonic.</param>
+    /// <param name="inputList">Input list of tuple mnemonic values.</param>
+    /// <param name="requestedGapSize">Requested gap size defined by user.</param>
+    /// <param name="isLogIncreasing">Is log direction increasing.</param>
+    /// <returns>Report items with gap size information.</returns>
+    private IEnumerable<AnalyzeGapReportItem> GetAnalyzeGapReportItem(string mnemonic, IList<Tuple<Index, string>> inputList, Index requestedGapSize, bool isLogIncreasing)
+    {
+        List<AnalyzeGapReportItem> gapValues = new();
+        Index startGapIndex = null;
+        bool insideGap = false;
+        
+        for (int i = 0; i < inputList.Count; i++)
+        {
+            var typeItem =  inputList[i].Item1;
+            string value = inputList[i].Item2;
+            if (string.IsNullOrEmpty(value))
```

Comment:
Checking string.IsNullOrEmpty(value) is incorrect as it will give false negatives way too often. Gaps should be instead detected by keeping track of the index of the last encountered value for the given mnemonic and compared with the index of the current value.

Reply:
For example, when analyzing only a single mnemonic, it would be impossible to detected any gaps using this method.

Reply:
What do you mean give false negatives way too often?

Reply:
fixed